### PR TITLE
Added the cart feature

### PR DESCRIPTION
**Pull request by karanvirchaudhary**

### CODE REVIEW

File: src/main/java/Sysc4806Group/demo/controllers/CartController.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import Sysc4806Group.demo.repositories.BookRepository;
+import Sysc4806Group.demo.repositories.CartRepository;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@Controller
+public class CartController {
```
Comment (abhiramj4) [inline src/main/java/Sysc4806Group/demo/controllers/CartController.java:14]: comments pls

**Comment by karanvirchaudhary** [comment]
I added the cart feature. The Cart model, entity and controller are all present. 
When we are in the user's home page (where they can access the bookstore, upload a book), they can view their cart. 
The cart is successfully implemented where It will tell the user if the cart is empty. 
I was also able to link a cart to a user and actually be able to add a book to it. The problem is that I always get a template parsing error after my post request.

File: src/main/java/Sysc4806Group/demo/controllers/BookController.java [hunk 2]
```diff
@@ -94,4 +92,11 @@ public String updateBook(@ModelAttribute Book book, Model model) {
         repository.save(bookFromDB);
         return "view-book";
     }
+
+    @GetMapping("/addItem/{id}")
+    public String addItemForm(@PathVariable String id, Model model) throws Exception {
```
Comment (dhyanpathak) [inline src/main/java/Sysc4806Group/demo/controllers/BookController.java]: I forgot to remove this.....pls remove this

**Review by dhyanpathak — CHANGES_REQUESTED** [review]

**Review by SVD6 — APPROVED** [review]
Looks good

File: src/main/java/Sysc4806Group/demo/controllers/BookController.java (hunk 2 above)
Comment (karanvirchaudhary) [reply src/main/java/Sysc4806Group/demo/controllers/BookController.java]: Hey Dhyan, 
I deleted those lines! Gotchu!

**Review by dhyanpathak — APPROVED** [review]

**Pull request merged**